include/sound/sof/ext_manifest4.h

```diff
@@ -0,0 +1,135 @@
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ */
+
+/*
+ * Extended manifest is a place to store metadata about firmware, known during
+ * compilation time - for example firmware version or used compiler.
+ * Given information are read on host side before firmware startup.
+ * This part of output binary is not signed.
+ */
+
+#ifndef __SOF_FIRMWARE_EXT_MANIFEST4_H__
+#define __SOF_FIRMWARE_EXT_MANIFEST4_H__
+
+/* In ASCII  $AE1 */
+#define SOF_EXT_MAN4_MAGIC_NUMBER    0x31454124
+
+#define MAX_MODULE_NAME_LEN     8
+#define MAX_FW_BINARY_NAME      8
+#define DEFAULT_HASH_SHA256_LEN 32
+#define SOF_MAN4_FW_HDR_OFFSET	0x2000
+
+/*********************************************************************
+ *     css_manifest hdr
+ *-------------------
+ *     offset reserved for future
+ *-------------------
+ *     fw_hdr
+ *-------------------
+ *     module_entry[0]
+ *-------------------
+ *     module_entry[1]
+ *-------------------
+ *     ...
+ *-------------------
+ *     module_entry[n]
+ *-------------------
+ *     FW content
+ *-------------------
+ *********************************************************************/
+
+struct sof_ext_manifest4_hdr {
+	u32 id;
+	u32 len;
+	u16 version_major;
+	u16 version_minor;
+	u32 entries;
+} __packed;
+
+struct sof_man4_fw_img_flags {
+	uint32_t _rsvd0 : 31;
+	uint32_t tp : 1;
+} __packed;
+
+struct sof_man4_fw_binary_header {
+	/* This part must be unchanged to be backward compatible with SPT-LP ROM */
+	uint32_t		id;
+	uint32_t		len; /* sizeof(sof_man4_fw_binary_header) in bytes */
+	uint8_t			name[MAX_FW_BINARY_NAME];
+	uint32_t		preload_page_count; /* number of pages of preloaded image */
+	struct sof_man4_fw_img_flags	fw_image_flags;
+	uint32_t		feature_mask;
+	uint16_t		major_version;
+	uint16_t		minor_version;
+	uint16_t		hotfix_version;
+	uint16_t		build_version;
+	uint32_t		num_module_entries;
+
+	/* This part may change to contain any additional data for BaseFw that is skipped by ROM */
+	uint32_t		hw_buf_base_addr;
+	uint32_t		hw_buf_length;
+	uint32_t		load_offset; /* This value is used by ROM */
+} __packed;
+
+struct sof_man4_module_type {
+	uint32_t load_type : 4; /* MT_BUILTIN, MT_LOADABLE */
+	uint32_t auto_start : 1; /* 0 - manually created, 1 - created by Module Manager */
+	uint32_t domain_ll : 1; /* support LL domain */
+	uint32_t domain_dp : 1; /* support DP domain */
+	uint32_t lib_code : 1; /* determines if module is place holder for common library code */
+	uint32_t _rsvd : 24;
+} __packed;
+
+struct sof_man4_segment_flags {
+	uint32_t	contents : 1;
+	uint32_t	alloc : 1;
+	uint32_t	load : 1;
+	uint32_t	readonly : 1;
+	uint32_t	code : 1;
+	uint32_t	data : 1;
+	uint32_t	_rsvd0 : 2;
+
+	uint32_t	type : 4;
+	uint32_t	_rsvd1 : 4;
+
+	uint32_t	length : 16; /* segment length in pages */
```

Comment:
does the kernel need to know? That looks more of a uapi between userpace (rimage) and firmware.